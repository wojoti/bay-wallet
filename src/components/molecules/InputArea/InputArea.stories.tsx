import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import InputArea from './InputArea';

export default {
  title: 'Molecules/InputArea',
  component: InputArea,
  argTypes: {onChange: {action: 'onChange'}},
} as ComponentMeta<typeof InputArea>;
const Template: ComponentStory<typeof InputArea> = args => (
  <InputArea {...args} />
);

export const Username = Template.bind({});
Username.args = {label: 'username', testId: ''};

export const Password = Template.bind({});
Password.args = {label: 'password', testId: ''};
