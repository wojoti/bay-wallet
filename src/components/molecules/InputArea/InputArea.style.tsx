import {StyleSheet} from 'react-native';
import {secondary} from 'theme/colors';

const style = StyleSheet.create({
  fixedLabel: {
    flex: 0,
    width: 95,
    marginHorizontal: 15,
    paddingHorizontal: 10,
    backgroundColor: secondary,
    alignItems: 'flex-start',
    top: 15,
    zIndex: 1,
  },
});
export default style;
