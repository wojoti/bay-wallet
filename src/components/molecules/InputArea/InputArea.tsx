import {Container, Input, Label} from 'components/atoms';
import {PropsWithChildren} from 'react';
import style from './InputArea.style';

export type InputAreaProps = PropsWithChildren<{
  type: 'username' | 'password' | 'phone' | 'password-confirm';
  onChange: (value: string) => void;
  testId?: string;
}>;

const InputArea = ({type, onChange, testId}: InputAreaProps) => {
  switch (type) {
    case 'username':
      return (
        <Container testId={testId} align={'flex-start'}>
          <Container customStyle={style.fixedLabel}>
            <Label testId="inputarea-label">Username</Label>
          </Container>
          <Input
            type={'username'}
            onChange={onChange}
            testId="inputarea-input"
          />
        </Container>
      );
    case 'password':
      return (
        <Container testId={testId} align={'flex-start'}>
          <Container customStyle={style.fixedLabel}>
            <Label testId="inputarea-label">Password</Label>
          </Container>
          <Input
            type={'password'}
            onChange={onChange}
            testId="inputarea-input"
          />
        </Container>
      );
    case 'phone':
      return (
        <Container testId={testId} align={'flex-start'}>
          <Container customStyle={style.fixedLabel}>
            <Label testId="inputarea-label">Phone</Label>
          </Container>
          <Input type={'phone'} onChange={onChange} testId="inputarea-input" />
        </Container>
      );
    case 'password-confirm':
      return (
        <Container testId={testId} align={'flex-start'}>
          <Container customStyle={style.fixedLabel}>
            <Label testId="inputarea-label">Confirm </Label>
          </Container>
          <Input
            type={'password-confirm'}
            onChange={onChange}
            testId="inputarea-input"
          />
        </Container>
      );
  }
};

export default InputArea;
