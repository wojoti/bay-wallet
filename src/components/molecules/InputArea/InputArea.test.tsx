import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import InputArea, {InputAreaProps} from './InputArea';

let value = '';
const handleChange = jest.fn(inputareaValue => (value = inputareaValue));
const props: InputAreaProps = {
  onChange: handleChange,
  testId: 'test-inputarea-id',
  type: 'username',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<InputArea {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render inputarea', () => {
  render(<InputArea {...props} />);
  const inputareaElement = screen.getByTestId('test-inputarea-id');
  expect(inputareaElement).toBeOnTheScreen();
});

test('should render inputarea - label', () => {
  render(<InputArea {...props} />);
  const inputareaElement = screen.getByTestId('test-inputarea-id');
  expect(inputareaElement).toBeOnTheScreen();
  const inputareaLabelElement =
    within(inputareaElement).getByTestId('inputarea-label');
  expect(inputareaLabelElement).toBeOnTheScreen();
});

test('should render inputarea - input', () => {
  render(<InputArea {...props} />);
  const inputareaElement = screen.getByTestId('test-inputarea-id');
  expect(inputareaElement).toBeOnTheScreen();
  const inputareaInputElement =
    within(inputareaElement).getByTestId('inputarea-input');
  expect(inputareaInputElement).toBeOnTheScreen();
});

test('should render inputarea - input - should handle onChange', () => {
  render(<InputArea {...props} />);
  const inputareaElement = screen.getByTestId('test-inputarea-id');
  expect(inputareaElement).toBeOnTheScreen();
  const inputareaInputElement =
    within(inputareaElement).getByTestId('inputarea-input');
  expect(inputareaInputElement).toBeOnTheScreen();
  fireEvent.changeText(inputareaInputElement, 'test1234');
  expect(handleChange).toHaveBeenCalledTimes(1);
  expect(value).toBe('test1234');
});
