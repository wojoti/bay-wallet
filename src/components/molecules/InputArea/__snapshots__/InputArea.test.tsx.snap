// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-start",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="test-inputarea-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "alignItems": "flex-start",
          "backgroundColor": "white",
          "flex": 0,
          "marginHorizontal": 15,
          "paddingHorizontal": 10,
          "top": 15,
          "width": 95,
          "zIndex": 1,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#363062",
          "fontFamily": "Poppins-Light",
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "left",
        }
      }
      testID="inputarea-label"
    >
      Username
    </Text>
  </View>
  <TextInput
    accessibilityHint="Allow to insert username"
    accessibilityLabel="Login input field"
    autoComplete="username"
    autoCorrect={false}
    onChangeText={[MockFunction]}
    style={
      {
        "borderColor": "#363062",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontFamily": "Poppins-Light",
        "fontSize": 16,
        "fontWeight": "400",
        "height": 65,
        "paddingLeft": 30,
        "width": "100%",
      }
    }
    testID="inputarea-input"
  />
</View>
`;
